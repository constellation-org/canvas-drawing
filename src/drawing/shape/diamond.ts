import { ShapeToDraw, drawShape } from './shape'

export const drawDiamond = (
  context: CanvasRenderingContext2D,
  args: ShapeToDraw
) => {
  const path = (context: CanvasRenderingContext2D, args: ShapeToDraw) => {
    context.beginPath()
    context.moveTo(args.x + args.width / 2, args.y)
    context.lineTo(args.x + args.width, args.y + args.height / 2)
    context.lineTo(args.x + args.width / 2, args.y + args.height)
    context.lineTo(args.x, args.y + args.height / 2)
    context.closePath()
  }
  drawShape(context, path, args)
}
