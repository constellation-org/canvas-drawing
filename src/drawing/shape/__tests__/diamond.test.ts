import { drawDiamond } from '../diamond'

describe('[Diamond]', () => {
  let canvas: HTMLCanvasElement
  let context: CanvasRenderingContext2D
  beforeEach(function () {
    canvas = document.createElement('canvas')
    canvas.setAttribute('width', "100")
    canvas.setAttribute('height', "100")
    context = canvas.getContext('2d') as CanvasRenderingContext2D
  })
  it('should draw diamond on top left corner', () => {
    drawDiamond(context, { x: 0, y: 0, width: 20, height: 20})
    expect(canvas.toDataURL('image/png')).toMatchSnapshot()
  })
  it('should draw diamond on bottom right corner', () => {
    drawDiamond(context, { x: 80, y: 80, width: 20, height: 20})
    expect(canvas.toDataURL('image/png')).toMatchSnapshot()
  })
  it('should draw diamond in the middle', () => {
    drawDiamond(context, { x: 30, y: 40, width: 40, height: 20})
    expect(canvas.toDataURL('image/png')).toMatchSnapshot()
  })
})
