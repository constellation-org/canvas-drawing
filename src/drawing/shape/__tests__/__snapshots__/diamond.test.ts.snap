// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Diamond] should draw diamond in the middle 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGQAAABkCAYAAABw4pVUAAAAAXNSR0IArs4c6QAAAlBJREFUeF7tmUFKw2AYBV8P4tpriHcQRE/iQkVdeBJF8A7iNVx7EQmkIKExf2rbzKfTjYvW9nUmkwS6ig8UgRVqjWOiENhBoBCFwAjA5liIQmAEYHMsRCEwArA5FqIQGAHYHAtRCIwAbI6FKARGADbHQhQCIwCbYyEKgRGAzbEQhcAIwOZYiEJgBGBzLEQhMAKwORaiEBgB2BwLUQiMAGyOhSgERgA2x0IUAiMAm2MhCoERgM2xEIXACMDmWIhCYARgcyxEITsncJLktn/XuyTvO/+EA75h5ULWIrq/3x+dkLJiKgoZEzE8jkuKqSSkVURpMRWEtIh46S2c/3C6L1EMWUiLiOck90k+ehHHSW6SXFQVQxSyjYgh/7JiSEJaRDwlefhWxNQNaTkxBCH7EFG2mCWFtIgYXiOmiph6Hl8MWchnkrckjzNOUS1CrpKcJjkaefGid2NLClnzaCll7rVj0ynrOskl/e6LIGSfYrpTVAkRawgkIXPETF1b8NeKsVKJQn4jpqwIciHDg6flGvPa/9MZ/RoxdddBLmQbMZu+76J3TVMChs9XEjLnVNa9tpSISqessYPMH6jm5neg1/sT7oFA/8uPqXgN+dOiFALTqxCFwAjA5liIQmAEYHMsRCEwArA5FqIQGAHYHAtRCIwAbI6FKARGADbHQhQCIwCbYyEKgRGAzbEQhcAIwOZYiEJgBGBzLEQhMAKwORaiEBgB2BwLUQiMAGyOhSgERgA2x0IUAiMAm2MhCoERgM2xEIXACMDmWIhCYARgcywEJuQLMmpbZYCXqywAAAAASUVORK5CYII="`;

exports[`[Diamond] should draw diamond on bottom right corner 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGQAAABkCAYAAABw4pVUAAAAAXNSR0IArs4c6QAAAkdJREFUeF7tmdtJxFAABc+24KMEEaxB1A4sSe3IEkSswQd24KsGCWRhSW7iktwr8zH++OPmHmZ2spLdxB8UgQ1qjWOiENibQCEKgRGAzbEQhcAIwOZYiEJgBGBzLEQhMAKwORaiEBgB2BwLUQiMAGyOhSgERgA2x0IUAiMAm2MhCoERgM2xEIXACMDmWIhCYARgcyxEITACsDkWohAYAdgcC1EIjABsjoUoBEYANsdCFAIjAJtjIQqBEYDNsRCFwAjA5liIQmAEYHMsRCEwArA5FqIQGAHYHAtRCIwAbI6FKARGADbHQhQCIwCbYyEKgRGAzbEQhcAIwOZYiEJgBGBzLEQhMAKwORaiEBgB2BwLUQiMAGyOhSgERgA2x0IUAiMAm2MhCoERgM2xEIXACMDmWIhCYARgcyxEITACsDkWohAYAdgcC1EIjABsjoUoBEbgf+bc9sdsf0+eaiHthXQSbvpj7pLMSlFIWyG7MrYnzUpRSDshJRl/SlFIGyFzMmalKKS+kJKMn/6Yg8Fxo9uXQuoKKcn4SnLeH/OU5GhOikLqCSnJ+E5ykeS5P+YsyWOSwykpCqkjZK6Mt8ERp0kmS1HIeiElGZ9JrnbKGJ7SlfKQ5HhYikLaCHlPcp3kZeLynZD7JCcKWS+gdAVvWW24rrpqScpHksskrzsf6sVb1faRiresVQ5GL/bf3ro8q1xtqpTuzT/6EB8+bLSQKg72KmX4R8WHjAppI6S7qg8X27FdfGUfvy9G1+6FfkHVju3iK+/9Fe4vyB49ZSmmlsEAAAAASUVORK5CYII="`;

exports[`[Diamond] should draw diamond on top left corner 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGQAAABkCAYAAABw4pVUAAAAAXNSR0IArs4c6QAAAshJREFUeF7tmNuN01AABc+2wKMEhLQ1IKADSgI6ooQVogYeogNeNawsJZJlOfF1zpiv2Z/83DOWZmQn67uM/X04HTt/jq08tdvA3cBiivD+dO5jEqMMSLv1yFaQeYzzNYxyq+2B3bUgazGMMiC1OXIpyLUYRmmMb2zXgqzF+HfiPFnwfHzBcZZB1mL8SfLqdN0vSZ4ZBa4ww82DrMX4m+R1kq+nzX2Sz0meGuWYKOcg1+6MH4tLv0zinXJMj0xB1mL8TvJ2dmcsLz/dKQ9JnnunsGUuBfmZ5F2SbxcuNwX5lOSFQfggE9FHFuv1ZtrWl/qvJG+SfJ99qfuouln39tCfvduO/uuJ0X8MpztlOuuX+MF5fHVysOC9eF8u7jV28Hlfvx8seC9+K8jyJ7EvE/ca3nl+JMg5yvxz52U8PmpgNMgoz3OlAYOUAum5QWijJc8gpUB6bhDaaMkzSCmQnhuENlryDFIKpOcGoY2WPIOUAum5QWijJc8gpUB6bhDaaMkzSCmQnhuENlryDFIKpOcGoY2WPIOUAum5QWijJc8gpUB6bhDaaMkzSCmQnhuENlryDFIKpOcGoY2WPIOUAum5QWijJc8gpUB6bhDaaMkzSCmQnhuENlryDFIKpOcGoY2WPIOUAum5QWijJc8gpUB6bhDaaMkzSCmQnhuENlryDFIKpOcGoY2WPIOUAum5QWijJc8gpUB6bhDaaMkzSCmQnhuENlryDFIKpOcGoY2WPIOUAum5QWijJc8gpUB6bhDaaMkzSCmQnhuENlryDFIKpOcGoY2WPIOUAum5QWijJc8gpUB6bhDaaMkzSCmQnhuENlryDFIKpOcGoY2WPIOUAum5QWijJc8gpUB6bhDaaMkzSCmQnhuENlryDFIKpOcGoY2WPIOUAum5QWijJc8gpUB6bhDaaMkzSCmQnhuENlryDFIKpOePU2Y+ZTn6ge8AAAAASUVORK5CYII="`;
